src/backend/core/services/marketing_service.py

```diff
@@ -0,0 +1,105 @@
+"""Marketing service in charge of pushing data for marketing automation."""
+
+import logging
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+import brevo_python
+
+logger = logging.getLogger(__name__)
+
+
+class ContactCreationError(Exception):
+    """Raised when the contact creation fails."""
+
+
+@dataclass
+class ContactData:
+    """Data structure for contact information."""
+
+    email: str
+    attributes: Optional[Dict[str, str]] = None
+    list_ids: Optional[List[int]] = None
+    update_enabled: bool = True
+
+
+class MarketingService:
```

Comment:
MarketingService should probably be a BaseMarketingService base class which you derive as BrevoMarketingService, one for every backend that people will want to support? I shouldn't be more work.